### ISSUE TITLE
URL encoding link to map files 

### ISSUE THREAD

**Issue by sorskoot** [evanw/esbuild#3982]
### Issue

When a space is present in the filename of a Javascript file, this space also appears in the link to the map file at the end of a bundle. 
Like `//# sourceMappingURL=Some Random File.js.map `. This will result in the map file not being found. 

### Expected behavior
By encoding it correctly there are now actual spaces anymore and the browser can show the correct file
`//# sourceMappingURL=Some%20Random%20File.js.map `

**Comment by evanw** [comment]
I agree that this should be fixed. However, it sounds like something that could break existing code that finds the source mapping URL in esbuild's output using regular expressions and string manipulation. So I'm going to wait until the next breaking change release to do this.
